lib/logstash/filters/csv.rb

```diff
@@ -91,5 +98,44 @@ def filter(event)
 
   end # def filter
 
+  private
+
+  def ignore_field?(index)
+    !@columns[index] && !@autogenerate_column_names
+  end
+
+  def should_convert?(field_name)
+    !@convert[field_name].nil?
+  end
+
+  def convert(field_name, value)
+    transformation = @convert[field_name].to_sym
+    converters[transformation].call(value)
+  end
+
+  def converters
+    @converters ||= {
+      :integer => lambda do |value|
+        CSV::Converters[:integer].call(value)
+      end,
+      :float => lambda do |value|
+        CSV::Converters[:float].call(value)
+
+      end,
+      :date => lambda do |value|
+        CSV::Converters[:date].call(value)
+
+      end,
+      :date_time => lambda do |value|
+        CSV::Converters[:date_time].call(value)
```

Comment:
good catch, I amended this situation by adding a validation in the register method, so there is actually no option to pass a wrong validation here.